master/buildbot/util/service.py

```diff
@@ -186,6 +186,8 @@ class AsyncService(service.Service):
     def setServiceParent(self, parent):
         if self.parent is not None:
             yield self.disownServiceParent()
+        # XXX(sa2ajj): why this magic?  The twisted version of setServiceParent
+        # does not lookg this complicated (the line below)
         parent = service.IServiceCollection(parent, parent)
```

Comment:
Actually, it has [exactly that line](http://twistedmatrix.com/trac/browser/tags/releases/twisted-14.0.0/twisted/application/service.py#L184)